monitorcontroller/monitorcontroller.go

```diff
@@ -98,16 +106,81 @@ func (monitor *MonitorController) Close() {
 	}
 }
 
-// SendMonitoringData sends monitoring data.
-func (monitor *MonitorController) SendMonitoringData(monitoringData cloudprotocol.NodeMonitoringData) {
+// SendNodeMonitoring sends monitoring data.
+func (monitor *MonitorController) SendNodeMonitoring(nodeMonitoring aostypes.NodeMonitoring) {
 	monitor.Lock()
-	defer monitor.Unlock()
 
-	if len(monitor.monitoringQueue) >= monitor.monitoringQueueSize {
-		monitor.monitoringQueue = monitor.monitoringQueue[1:]
+	// calculate size of input parameter
+	messageSize := 0
+
+	message, err := json.Marshal(nodeMonitoring)
+	if err == nil {
+		messageSize = len(message)
+	} else {
+		log.Errorf("Can't marshal nodeMonitoring: %v", err)
+	}
+
+	forceSendOfflineMessages := monitor.currentMessageSize+messageSize > monitor.maxMessageSize &&
```

Comment:
We should send immediately and do  not wait when offline queue is full.

Reply:
changed